pom.data.xml

```diff
@@ -602,6 +602,7 @@
     </profile>
   </profiles>
 
+  <!--
```

Comment:
> There's an open question about the need for the additional test module entries but that only really a concern, for me, until it's understood why they're necessary. If the situation arises where someone is adding a new library and/or area and a similar issue pops up, needing a dependency where seemingly we shouldn't, I'd like to be able to explain why it's necessary.

This is a side effect of the ```-amd``` approach I was trying to use. I'll go through the PR and remove the ones that are unnecessary now that I don't need to force the dependency graph into a particular shape.

I'll also remove the commented out secitons.

Reply:
Wait ... I need to live those modules in there for some tooling that still depends on them being there. I'll do that instead. Glad you mentioned it.

Reply:
>> This is a side effect of the -amd approach
I recognize that but is this something we can predict? What's the pattern that causes us to need the extra entry and is it something we can effectively document/communicate with the Java team so everyone knows the deal and is on the same page?

Reply:
The most likely culprits are going to be things like identity, storage, perf-test-core. Hard to predict where it'll crop up because its a function of the dependencies that individual libraries take. I think it might happen more as we introduce new services that take dependencies on fundamental libraries like identity and storage.